### ISSUE TITLE
Feature request: disable built-in plugins

### ISSUE THREAD

**Issue by W1M0R** [moonrepo/proto#546]
### What version?

0.38.2

### Which command?

proto use

### What happened?

I'm trying to gradually enable proto-support for a  monorepo. The root of the monorepo contains the following `.prototools` file:

```toml
just = "1.30.0"

[plugins]
just = "https://raw.githubusercontent.com/Phault/proto-toml-plugins/main/just/plugin.toml"

[settings]
auto-install = true
detect-strategy = "only-prototools"
telemetry = false
```

When I run `proto use`, the command fails with the following:

```pwsh
> proto use --log trace
...
[TRACE 13:42:34.401] warpgate::plugin  Calling plugin function detect_version_files  id="go" plugin="93cecbf2-bee9-4ad8-865c-3affe097b6c3" input={}
[TRACE 13:42:34.402] warpgate::plugin  Called plugin function detect_version_files  id="go" plugin="93cecbf2-bee9-4ad8-865c-3affe097b6c3" output={"files":["go.mod","go.work"]}
[TRACE 13:42:34.403] proto_core::tool  Attempting to detect a version from directory  tool="go" dir="C:\\Users\\user2\\sdl"
[TRACE 13:42:34.405] starbase_utils::fs  Reading file  file="C:\\Users\\user2\\sdl\\go.work"
[TRACE 13:42:34.406] warpgate::plugin  Calling plugin function parse_version_file  id="go" plugin="93cecbf2-bee9-4ad8-865c-3affe097b6c3" input={"content":"go 1.19\r\n\r\nuse ./src/fdtsdlcli\r\nuse ./src/fdtsdltray\r\nuse ./src/pgsse\r\nuse ./labs/sdl-web/sdl-web-demo-1\r\nuse ./labs/sdl-web/sdl-web-demo-5/src-wails\r\nuse ./labs/fdtdb","file":"go.work"}
[TRACE 13:42:34.419] starbase::app  Running shutdown phase (because another phase failed) 
[TRACE 13:42:34.901] starbase_utils::net::offline  Resolving 142.251.47.206:80 
[TRACE 13:42:34.926] starbase_utils::net  Online! 
[TRACE 13:42:34.928] starbase_utils::fs  Reading file  file="C:\\Users\\user2\\.proto\\temp\\.last-version-check"
Error: plugin::call_func::failed

` in version string!r `
```

It looks like `proto use` is trying to parse version info from my `go.work` file. 

I mistakenly thought that `detect-strategy=only-prototools` would effectively only install `just`, but it is actually also processing the built-in plugins, including go in this case.

To reduce the potential for breaking changes to the project's dev environment, I don't want proto to do or install things that I did not explicitly ask for.

I can think of a few ways to achieve my goals:

1. Add a `detect-strategy=only-local-prototools` that can be interpreted by `proto use` to only install the explicit packages listed in the local `.prototools` file, potentially without resolving up the folder tree and without activating the built-in plugins.
2. Add a flag that would only use the local file, without resolving up the tree and without activating the built-in plugins, e.g. `proto use --local-only --no-builtins`.
3. Allow built-in plugins to be disabled in the `.prototools` file (I'd disable these in the root `.prototools` file of the monorepo, but then enable them incrementally as needed for `.prototools` files in subprojects of the monorepo), e.g:

```toml
[plugins]
go = false

# or all plugins:
[settings]
use-builtins = false
```

In the bigger picture, there might be an issue with the way the `go` plugin is trying to detect the version from my `go.work` file:

```go
go 1.19

use ./src/fdtsdlcli
use ./src/fdtsdltray
use ./src/pgsse
use ./labs/sdl-web/sdl-web-demo-1
use ./labs/sdl-web/sdl-web-demo-5/src-wails
use ./labs/fdtdb
```

For the case where proto fails to detect a version using an implicit built-in plugin (as above), it should not prevent `proto use` to complete successfully, i.e. those failures should just be ignored (maybe a `.prototools` setting or a `proto use` flag could explicitly request to ignore `proto use` failures.

If I delete the `go.work` file, then `proto use` succeeds. But I don't want to delete `go.work` since it is necessary for other parts of the environment.

The other workaround that is sufficient for my use case, is just to add a specific go version to the `.prototools` file, to prevent the automatic version detection from executing:

```toml
go = "1.19.13"
```

### Trace logs?

```pwsh
...
[TRACE 13:42:34.401] warpgate::plugin  Calling plugin function detect_version_files  id="go" plugin="93cecbf2-bee9-4ad8-865c-3affe097b6c3" input={}
[TRACE 13:42:34.402] warpgate::plugin  Called plugin function detect_version_files  id="go" plugin="93cecbf2-bee9-4ad8-865c-3affe097b6c3" output={"files":["go.mod","go.work"]}
[TRACE 13:42:34.403] proto_core::tool  Attempting to detect a version from directory  tool="go" dir="C:\\Users\\user2\\sdl"
[TRACE 13:42:34.405] starbase_utils::fs  Reading file  file="C:\\Users\\user2\\sdl\\go.work"
[TRACE 13:42:34.406] warpgate::plugin  Calling plugin function parse_version_file  id="go" plugin="93cecbf2-bee9-4ad8-865c-3affe097b6c3" input={"content":"go 1.19\r\n\r\nuse ./src/fdtsdlcli\r\nuse ./src/fdtsdltray\r\nuse ./src/pgsse\r\nuse ./labs/sdl-web/sdl-web-demo-1\r\nuse ./labs/sdl-web/sdl-web-demo-5/src-wails\r\nuse ./labs/fdtdb","file":"go.work"}
[TRACE 13:42:34.419] starbase::app  Running shutdown phase (because another phase failed) 
[TRACE 13:42:34.901] starbase_utils::net::offline  Resolving 142.251.47.206:80 
[TRACE 13:42:34.926] starbase_utils::net  Online! 
[TRACE 13:42:34.928] starbase_utils::fs  Reading file  file="C:\\Users\\user2\\.proto\\temp\\.last-version-check"
Error: plugin::call_func::failed

` in version string!r `
```

### Operating system?

Windows

### Architecture?

x64

**Comment by milesj** [comment]
I'll look into adding a setting for this in the next version.

As for the go parsing failure, that seems... odd. That code is very simple (https://github.com/moonrepo/go-plugin/blob/master/src/proto.rs#L52) and we even have tests for it (https://github.com/moonrepo/go-plugin/blob/master/tests/versions_test.rs#L77).

**Comment by milesj** [comment]
Landed in v0.39.
